proxy_test.go

```diff
@@ -121,14 +121,14 @@ func testHTTPConnect(t *testing.T, proxyURLModify func(*url.URL) *url.URL, proxy
 	recvBuf := make([]byte, len(msg))
 	done := make(chan struct{})
 	go func() {
+		defer close(done)
 		in, err := blis.Accept()
 		if err != nil {
 			t.Errorf("failed to accept: %v", err)
```

Comment:
While you are here, could you please switch this to be a `t.Fatalf()` instead of `t.Errorf()`. Thanks.

Reply:
Hi easwars, I think t.Fatalf() can't be used inside a goroutine created by the main test goroutine. According to https://golang.org/pkg/testing, "Fatalf is equivalent to Logf followed by FailNow" and "FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test. Calling FailNow does not stop those other goroutines".
Actually this is another checking rule of my static checker XD, otherwise I won't know this specific API behavior.

Reply:
How about I create a new channel `errch` to receive `err` of `in, err := blis.Accept()`, and change `<-done` at line 157 into a `select` with `case <-done: //do nothing` and `case err <-errch: t.Fatalf(err)` ?

Reply:
I like the idea of having an error channel. And I think if we use that, we dont need the `done` channel at all. We can send the `error` on it, if `blis.Accept()` returns an `error` and `nil` otherwise.

And at line 157, just block on the error channel, and if it returns a non-nil error, call `t.Fatalf`. Does this sound ok?

Reply:
And thanks for pointing out the behavior of `t.Fatalf`with goroutines.

Reply:
Sounds great to me. Yeah, we don't need another channel to do the job. 
Now we have `done := make(chan error)` to store `err` from `blis.Accept()`. At line 157, if `err` from `<-done` is not `nil`, `t.Fatalf` will be called.

Reply:
I squashed all the changes into one commit. The test function runs well in my computer. Please tell me if there are further problems.